ant-node/src/networking/record_store.rs

```diff
@@ -887,6 +887,11 @@ impl RecordStore for NodeRecordStore {
                     RecordKind::DataWithPayment(_) => {
                         debug!("Record {record_key:?} with payment shall always be processed.");
                     }
+                    RecordKind::DataWithMerklePayment(_) => {
```

Comment:
just to confirm, so this means we are going to support three types of payments:
* MerklePayment
* normal payment of 3
* normal payment of 1

right ?

even so, on Record level, there can be just one enum of DataWithPayment, then provide same payload to smart contract, and let smart contract using which detailed verification logic to handle ?